src/site/content/en/blog/sms-otp-form/index.md

```diff
@@ -0,0 +1,256 @@
+* **Account recovery**: When a user loses access to their account, there needs
+to be a way to recover it. Sending an email to their registered email address or
+an SMS OTP to their phone number are common account recovery methods.
+* **Payment confirmation**: In payment systems, some banks or credit card
+issuers request additional authentication from the payer for security reasons.
+SMS OTP is commonly used for that purpose.
+
+This post explains best practices to build an SMS OTP form for the above use
+cases.
+
+{% Aside 'caution' %}
+While this post discusses SMS OTP form best practices, be aware that SMS OTP is
+not the most secure method of authentication by itself. That's because phone
+numbers are known to be recycled and sometimes hijacked. And [the concept of OTP
+itself is not phishing resistant](https://youtu.be/kGGMgEfSzMw?t=1133).
+
+If you are looking for better security, consider using WebAuthn. You can learn
+more about it from the talk "[What's new in sign-up &
+sign-in](https://goo.gle/webauthn-video)" at the Chrome Dev Summit 2019.
+{% endAside %}
+
+## Checklist
+
+To provide the best user experience with the SMS OTP:
+
+* Use an `<input>` element with:
+    * `type="text"`
+    * `inputmode="numeric"`
+    * `autocomplete="one-time-code"`
+* Use `@${bound-domain} #${OTP}` as the last line of the SMS text message.
+* Use the Web OTP API.
+
+## Use the `<input>` element
+
+Using a form with an `<input>` element is the most important best practice you
+can follow as it will definitely work on any browser. Even if other items in
+this article don't work, the user can still enter and submit the OTP by doing it
+manually.
+
+```html
+<form action="/verify-otp" method="POST">
+  <input type="text"
+         inputmode="numeric"
+         autocomplete="one-time-code"
+         pattern="\d{6}"
+         required>
+</form>
+```
+
+The following are a few ideas to ensure an input field gets the best out of
+browser functionality.
+
+{% Aside 'gotchas' %}
+There are more general form best practices other than ones described below. [Sam
+Dutton](https://twitter.com/sw12)'s [Sign-in form best
+practices](https://web.dev/sign-in-form-best-practices/) is a great starting
+point.
+{% endAside %}
+
+### `type="text"`
+
+Because OTPs are usually 5 or 6 digit numbers, you may be tempted to use
+`type="number"` for an input field because it turns the mobile keyboard to be
+numbers only. However, it's not the best idea, because the browser expects an
+input field to be a countable number rather than a sequence of multiple numbers,
+and this causes unexpected behaviors. Using `type="number"` causes up and down
+buttons to be displayed beside the input field, and pressing these buttons
+increments or decrements the number and may remove preceding zeros.
+
+Use `type="text"` instead. This won't turn the mobile keyboard into numbers
+only, but that is fine because the next tip for using `inputmode="numeric"` does
+that job.
```

Comment:
```suggestion
Because OTPs are usually five or six digit numbers, using
`type="number"` for an input field might seem intuitive because it turns the mobile keyboard 
to be numbers only. This not recommended because the browser expects an
input field to be a countable number rather than a sequence of multiple numbers,
which can cause unexpected behavior. Using `type="number"` causes up and down
buttons to be displayed beside the input field; pressing these buttons
increments or decrements the number and may remove preceding zeros.

Use `type="text"` instead. This won't turn the mobile keyboard into numbers
only, but that is fine because the next tip for using `inputmode="numeric"` does
that job.
```